src/ert/run_models/everest_run_model.py

```diff
@@ -355,40 +355,68 @@ def _run_forward_model(
         self,
         control_values: NDArray[np.float64],
         model_realizations: list[int],
-        active_control_vectors: list[bool] | None = None,
+        active_sim_ids: list[bool] | None = None,
     ) -> tuple[NDArray[np.float64], NDArray[np.float64] | None, list[int]]:
         # Reset the current run status:
         self._status = None
 
+        n_simulations = len(control_values)
+        has_constraints = bool(self._everest_config.constraint_names)
+
         # Get cached_results:
-        cached_results = self._get_cached_results(control_values, model_realizations)
+        cached_objectives, cached_constraints = self._get_cached_results_by_sim_id(
+            control_values, model_realizations
+        )
 
         # Collect the indices of the controls that must be evaluated in the batch:
-        evaluated_control_indices = [
-            idx
-            for idx in range(control_values.shape[0])
-            if idx not in cached_results
-            and (active_control_vectors is None or active_control_vectors[idx])
+        sim_ids_to_evaluate = [
+            sim_id
+            for sim_id in range(control_values.shape[0])
+            if sim_id not in cached_objectives
+            and (active_sim_ids is None or active_sim_ids[sim_id])
         ]
 
+        if not sim_ids_to_evaluate:
+            objectives = np.array(
+                [cached_objectives.get(sim) for sim in range(n_simulations)],
+                dtype=np.float64,
+            )
+
+            constraints = (
+                np.array(
+                    [
+                        cached_constraints.get(
+                            sim,
+                        )
+                        for sim in range(n_simulations)
+                    ],
+                    dtype=np.float64,
+                )
+                if has_constraints
+                else None
+            )
+
+            # Increase batch ID, preserves current behavior
+            # Should be revised
+            self._batch_id += 1
+            return objectives, constraints, sim_ids_to_evaluate
+
```

Comment:
This should work fine. Maybe there is a way to consolidate this piece of code and the one below into a method call, maybe combining it with `_get_cached_results_by_sim_id` or something like that? Now these are big pieces of code, distracting a bit from the flow of the method. And the are very similar, apart from taking defaults from `simulated_constraints`.